src/main/scala/edu/berkeley/ce/rockslicing/Block.scala

```diff
@@ -28,12 +47,50 @@ case class Face(normalVec: (Double,Double,Double), distance: Double, phi: Double
     val newD = if (math.abs(d) > NumericUtils.EPSILON) d else 0.0
     val newPhi = if (math.abs(phi) > NumericUtils.EPSILON) phi else 0.0
     val newCohesion = if (math.abs(cohesion) > NumericUtils.EPSILON) cohesion else 0.0
-    Face((newA,newB,newC), newD, newPhi, newCohesion)
+    Face((newA,newB,newC), newD, newPhi, newCohesion, processorJoint)
   }
 }
 
 object Block {
   /**
+    * Compare two blocks for equality within specified tolerance
+    * @param block1 First input block
+    * @param block2 Second input block
+    * @return True if blocks are equal, otherwise false
+    */
+  def compareBlocks(block1: Block, block2: Block, tolerance: Double=NumericUtils.EPSILON):
+                    Boolean = {
+    val block1Centroid = (block1.centerX, block1.centerY, block1.centerZ)
+    val updatedBlock2 = Block(block1Centroid, block2.updateFaces(block1Centroid))
+    val sortedFaces1a = 
+      block1.faces.sortWith(_.a < _.a)
+    val sortedFaces1b = 
+      sortedFaces1a.sortWith(_.b < _.b)
+    val sortedFaces1c = 
+      sortedFaces1b.sortWith(_.c < _.c)
+    val sortedFaces1 = 
+      sortedFaces1c.sortWith(_.d < _.d)
+
+    val sortedFaces2a = 
+      updatedBlock2.faces.sortWith(_.a < _.a)
+    val sortedFaces2b = 
+      sortedFaces2a.sortWith(_.b < _.b)
+    val sortedFaces2c = 
+      sortedFaces2b.sortWith(_.c < _.c)
+    val sortedFaces2 = 
+      sortedFaces2c.sortWith(_.d < _.d)
+
+    val faces = sortedFaces1.zip(sortedFaces2)
+    val faceMatches = faces map { case (face1, face2) =>
+      Face.compareFaces(face1, face2)
+    }
+    (math.abs(block1.centerX - updatedBlock2.centerX) < tolerance) &&
+    (math.abs(block1.centerY - updatedBlock2.centerY) < tolerance) &&
+    (math.abs(block1.centerZ - updatedBlock2.centerZ) < tolerance) &&
+    (!faceMatches.contains(false))
```

Comment:
How about getting rid of the `map` on line 84 and instead just doing something like `faces forall { case (f1, f2) => Face.compareFaces(f1, f2)` ?